tinygrad/nn/__init__.py

```diff
@@ -81,3 +81,17 @@ def __call__(self, x:Tensor):
 
 class LayerNorm2d(LayerNorm):
   def __call__(self, x): return super().__call__(x.permute(0, 2, 3, 1)).permute(0, 3, 1, 2)
+
+class Embedding:
+  def __init__(self, vocab_size:int, embed_size:int):
+    self.vocab_counter = Tensor.arange(vocab_size, requires_grad=False)
+    self.weight = Tensor.glorot_uniform(vocab_size, embed_size)
+
+  def __call__(self, idx:Tensor) -> Tensor:
+    oha = []
+    for i in range(idx.shape[0]):
```

Comment:
Err, this is really slow. Look at how I do it in llama.

Reply:
Maybe `Tensor.arange(vocab_size, requires_grad=False).reshape(1, 1, vocab_size).expand(*idx.shape, vocab_size) @ self.weight`